src/phylum/ci/ci_base.py

```diff
@@ -181,6 +184,30 @@ def depfiles(self) -> Depfiles:
             self.returncode = ReturnCode.NO_DEPFILES_PROVIDED
         raise SystemExit(self.returncode)
 
+    def _exclude_depfiles(self, provided_depfiles: DepfileEntries) -> DepfileEntries:
+        """Apply exclusion patterns to provided dependency files and return the remaining ones."""
+        arg_exclusions: Optional[list[list[str]]] = self.args.exclude
+        if not arg_exclusions:
+            LOG.debug("No dependency file exclusion patterns provided.")
+            return provided_depfiles
+
+        # flatten the list of lists
+        provided_arg_exclusions = list(chain.from_iterable(arg_exclusions))
+        LOG.debug("Exclusion patterns provided as arguments: %s", provided_arg_exclusions)
+
+        excluded_depfiles: DepfileEntries = []
+        for depfile in provided_depfiles:
+            for exclusion in provided_arg_exclusions:
+                if depfile.path.match(exclusion):
```

Comment:
The `match()` method supports absolute patterns. It might be nice to make `depfile` absolute here in order to take advantage of this feature.

For example, this would allow a user to write `/requirements.txt` to exclude the `requirements.txt` file at the root of the repo without excluding any other `requirements.txt` files in the repo.

Reply:
`depfile` is already absolute and presents a challenge for matching because the full absolute path is likely not known ahead of time. For instance, the `/requirements.txt` exclusion example would need to be `/Users/maxrake/dev/phylum/phylum-ci/requirements.txt` in my local test case for it to match. Knowing the full path in CI environments is less likely.

However, I've come up with a solution to create a fake absolute path "rooted" at the working directory (presumably the project root) to allow for matching absolute patterns. Testing it and updating the docs is still ongoing, but a change is coming.

Reply:
Oops...didn't see the `EDIT:` part of the original comment until after posting a reply. It seems we were on the same page about it, though.